src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -142,6 +143,29 @@ public function testCanDecorateServiceSubscriber()
         $this->assertInstanceOf(DecoratedServiceSubscriber::class, $container->get(ServiceSubscriberStub::class));
     }
 
+    public function testCanDecorateServiceLocator()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('foo', 'stdClass')->setPublic(true);
+
+        $container->register(ServiceLocator::class)
+            ->addTag('container.service_locator')
+            ->setArguments([[new Reference('foo')]])
+        ;
+
+        $container->register(DecoratedServiceLocator::class)
+            ->setDecoratedService(ServiceLocator::class)
+            ->setPublic(true)
+            ->setArguments([new Reference(DecoratedServiceLocator::class.'.inner')])
+        ;
+
+        $container->compile();
+
+        $this->assertInstanceOf(DecoratedServiceLocator::class, $container->get(DecoratedServiceLocator::class));
```

Comment:
shouldnt this be `get(ServiceLocator::class)`?

Reply:
I think it doesn't matter as the test is interested in whether container was compiled without errors, not if decoration works